packages/geo/src/lib/search/shared/sources/source.ts

```diff
@@ -83,6 +84,37 @@ export class SearchSource {
     return this.options.params === undefined ? {} : this.options.params;
   }
 
+  /**
+   * Search settings
+   */
+  get settings(): SearchSourceSettings[] {
+    return this.options.settings === undefined ? [] : this.options.settings;
+  }
+
+  setParamFromSetting(setting) {
+      switch (setting.type) {
+        case 'radiobutton':
+          setting.values.forEach( conf => {
+            if (conf.enabled) {
+              this.options.params = Object.assign( (this.options.params || {}),
+                                                    { [setting.name] : conf.value } );
+            }
+          });
+          break;
+        case 'checkbox':
+          let confValue = '';
```

Comment:
You could do something like this. You wouldn't have to remove the tailing comma.

```
const values = [];
values.push(conf.value);
const value = values.join(',');
```

or
```
const value = setting.values
  .filter((conf) => conf.enabled)
  .map((conf) => conf.value)
  .join(',')
```